clarifai/models/model_serving/models/default_test.py

```diff
@@ -0,0 +1,230 @@
+    """ preprocess if input is image """
+    if "image" in self.triton_model_input_name:
+      h, w, _ = self.triton_model_config.input[0].dims
+      if h > -1 and w > -1:
+        import cv2
+
+        def _f(x):
+          logging.info(f"Preprocess reshape image => {(w, h, 3)}")
+          return cv2.resize(x, (w, h))
+
+    return None
+
+  def intitialize(
+      self,
+      model_type: str,
+      repo_version_dir: str,
+      is_instance_kind_gpu: bool = True,
+  ):
+    import sys
+    sys.path.append(repo_version_dir)
+    self.model_type = model_type
+    self.is_instance_kind_gpu = is_instance_kind_gpu
+    logging.info(self.model_type)
+    from model import TritonPythonModel
+
+    # Construct TritonPythonModel object
+    self.triton_model = TritonPythonModel()
+    self.triton_model.initialize(
+        dict(
+            model_repository=os.path.join(repo_version_dir, ".."),
+            model_instance_kind="GPU" if self.is_instance_kind_gpu else "cpu"))
+    # Get default config of model and model_type
+    self.default_triton_model_config = TritonModelConfig(
+        model_name=self.model_type,
+        model_version="1",
+        model_type=self.model_type,
+        image_shape=[-1, -1])
+    # Get current model config
+    self.triton_model_config = self.triton_model.config_msg
+    self.triton_model_input_name = self.triton_model.input_name
+    self.preprocess = self._get_preprocess()
+    # load labels
+    self._required_label_model_types = [
+        "visual-detector", "visual-classifier", "text-classifier", "visual-segmenter"
+    ]
+    self.labels = []
+    if self.model_type in self._required_label_model_types:
+      with open(os.path.join(repo_version_dir, "../labels.txt"), 'r') as fp:
+        labels = fp.readlines()
+        if labels:
+          self.labels = [line for line in labels if line]
+
+  def test_triton_config(self):
+    """ test Triton config"""
+    # check if input names are still matched
+    self.assertEqual(
+        self.triton_model_input_name, self.default_triton_model_config.input[0].name,
+        "input name of current model vs generated model must be matched "
+        f"{self.triton_model_input_name} != {self.default_triton_model_config.input[0].name}")
+    # check if output names are still matched
+    default_output_names = [each.name for each in self.default_triton_model_config.output]
+    for output_name in self.triton_model_config.output:
+      self.assertIn(output_name.name, default_output_names,
+                    "output name of current model vs generated model must be matched "
+                    f"{output_name.name} not in {default_output_names}")
+
+  def test_having_labels(self):
+    if self.model_type in self._required_label_model_types:
+      self.assertTrue(
+          len(self.labels),
+          f"`labels.txt` is empty!. Model type `{self.model_type}` requires input labels in `labels.txt`"
+      )
+
+  def test_inference_with_predefined_inputs(self):
```

Comment:
This is a good test, but I think it may be difficult for the user to debug, for two reasons:
* it's in a library class, not the test file visible to the user, so when it fails they may need to look at the package code to debug
* it's cased with lots of input types other than the one that user is using, making most of the code irrelevant to their model

Another option could be to create a different test class with this test function for each model type, in separate files.  Then when generating the test repo, copy the class for the model type into the test module, so the test code is easily visible and editable (may or may not want to do this second part depending on what it looks like).

Not needed to merge, we could change that later if needed.

Reply:
Thank you for your suggestions. I believe users can follow the messages in the test and fix their inference code because it has messages for individual assertions.